src/live_restore/live_restore_fs.c

```diff
@@ -298,55 +300,76 @@ __live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_sessi
     /* Get files from destination. */
     WT_ERR(__live_restore_fs_backing_filename(
       &lr_fs->destination, session, lr_fs->destination.home, directory, &path_dest));
-    WT_ERR_ERROR_OK(lr_fs->os_file_system->fs_directory_list(
-                      lr_fs->os_file_system, wt_session, path_dest, prefix, &dirlist_dest, countp),
-      ENOENT, false);
 
-    for (namep = dirlist_dest; namep != NULL && *namep != NULL; namep++)
-        if (!WT_SUFFIX_MATCH(*namep, WT_LIVE_RESTORE_FS_TOMBSTONE_SUFFIX)) {
-            WT_ERR(__wt_realloc_def(session, &dirallocsz, count_dest + 1, &entries));
-            WT_ERR(__wt_strdup(session, *namep, &entries[count_dest]));
-            ++count_dest;
+    WT_ERR(lr_fs->os_file_system->fs_exist(
+      lr_fs->os_file_system, wt_session, path_dest, &dest_folder_exists));
 
-            if (single)
-                goto done;
-        }
+    if (dest_folder_exists) {
```

Comment:
Calling `WT_ERR_ERROR_OK` and ignoring `ENOENT` returns codes will work, but there's a problem when we list the contents of a directory present in source but not yet copied to the destination. This will cause the destination's `directory_list` to print error messages about a missing directory even though it's available and accessible by WiredTiger.

Instead of accepting `ENOENT` we first check for the presence of the directory, and only if it exists run the directory list. This also means we can properly return `ENOENT` when the directory is missing in the source and destination.